src/vm/gdbjit.cpp

```diff
@@ -1333,7 +1345,12 @@ void NotifyGdb::MethodCompiled(MethodDesc* MethodDescPtr)
     method = new FunctionMember(MethodDescPtr, locals.size, nArgsCount);
     // method return type
     method->m_member_type = GetArgTypeInfo(MethodDescPtr, pTypeMap, 0);
-    method->GetLocalsDebugInfo(pTypeMap, locals, symInfo[0].nativeOffset);
+
+    unsigned int firstLineIndex = 0;
+    for (;firstLineIndex < symInfoLen; firstLineIndex++) {
+        if (symInfo[firstLineIndex].lineNumber != 0 && symInfo[firstLineIndex].lineNumber != HiddenLine && symInfo[firstLineIndex].fileName[0] != 0) break;
+    }
+    method->GetLocalsDebugInfo(pTypeMap, locals, symInfo[firstLineIndex].nativeOffset);
```

Comment:
It seems that if the WideCharToMultiByte above failed to convert the filename (e.g. due to the target buffer not being large enough), the fileName[0] can be 0 even for a filename that was valid in the wide char form and that this loop can finish without finding an entry. 
Then the firstLineIndex would be symInfoLen and you would index behind the end of the symInfo.
So you'll need to add a check here.
I also wonder if the case when the symInfo[xxx].fileName[0] is zero due to a failed conversion of the filename can  cause issues at other places.